src/transaction.rs

```diff
@@ -159,31 +164,60 @@ pub struct JSTransaction {
     pub nullifiers: Vec<String>,
 }
 
-#[wasm_bindgen]
-pub fn create_transaction(
-    notes: JsValue,
-    extsk: &str,
-    to_address: &str,
-    change_address: &str,
+#[derive(Serialize, Deserialize)]
+pub struct Utxo {
+    txid: String,
+    vout: u32,
+    amount: u64,
+    private_key: Vec<u8>,
+    script: Vec<u8>,
+}
+
+#[derive(Serialize, Deserialize)]
+pub struct JSTxOptions {
+    notes: Option<Vec<(Note, String)>>,
+    utxos: Option<Vec<Utxo>>,
+    extsk: String,
+    to_address: String,
+    change_address: String,
     amount: u64,
     block_height: u32,
     is_testnet: bool,
-) -> JsValue {
-    // Note, witness
-    let mut notes = serde_wasm_bindgen::from_value::<Vec<(Note, String)>>(notes)
-        .expect("Cannot deserialize notes");
-    notes.sort_by_key(|(note, _)| note.value().inner());
-    let extsk = decode_extsk(extsk, is_testnet);
+}
+
+#[wasm_bindgen]
+pub fn create_transaction(options: JsValue) -> JsValue {
+    let JSTxOptions {
+        notes,
+        extsk,
+        to_address,
+        change_address,
+        amount,
+        block_height,
+        is_testnet,
+        utxos,
+    } = serde_wasm_bindgen::from_value::<JSTxOptions>(options).expect("Cannot deserialize notes");
+
+    let extsk = decode_extsk(&extsk, is_testnet);
     let network = if is_testnet {
         Network::TestNetwork
     } else {
         Network::MainNetwork
     };
+    let input = if let Some(mut notes) = notes {
```

Comment:
add in documentation that notes have priority? or throw error if user insert both utxo and notes?

Reply:
Added an assert!